Source/Core/DolphinQt/MainWindow.cpp

```diff
@@ -357,24 +357,26 @@ void MainWindow::ShutdownControllers()
 
 void MainWindow::InitCoreCallbacks()
 {
-  connect(&Settings::Instance(), &Settings::EmulationStateChanged, this, [=](Core::State state) {
-    if (state == Core::State::Uninitialized)
-      OnStopComplete();
-    if (state != Core::State::Uninitialized && NetPlay::IsNetPlayRunning() && m_controllers_window)
-      m_controllers_window->reject();
-
-    if (state == Core::State::Running && m_fullscreen_requested)
-    {
-      FullScreen();
-      m_fullscreen_requested = false;
-    }
-  });
+  connect(&Settings::Instance(), &Settings::EmulationStateChanged, this,
+          [=, this](Core::State state) {
+            if (state == Core::State::Uninitialized)
+              OnStopComplete();
+            if (state != Core::State::Uninitialized && NetPlay::IsNetPlayRunning() &&
+                m_controllers_window)
+              m_controllers_window->reject();
+
+            if (state == Core::State::Running && m_fullscreen_requested)
+            {
+              FullScreen();
+              m_fullscreen_requested = false;
+            }
+          });
```

Comment:
We should create MainWindow::OnEmulationStateChange and put the contents of this lambda there.